### ISSUE TITLE
Add the possibility of putting notes more than the length of a pattern

### ISSUE THREAD

**Issue by TheMelancholyOfGigi198** [YuriSizov/boscaceoil-blue#28]
It would be fantastic to add the possibility of adding longer notes, as in the old Bosca Ceoil. For example, being able to put a 32 long note even if the pattern is 16 long. It would be very useful for putting background tools that accompany the song, for example, without putting a pattern to be 32 etc...

Old Bosca Ceoil example
![2024-07-14_14-36](https://github.com/user-attachments/assets/251e88e5-a4c4-4961-97ec-5b452d8d0bf2)

**Comment by YuriSizov** [comment]
Original Bosca doesn't actually have an upper limit for the note length, but that doesn't seem to be very usable. What would be a reasonable limit, in your opinion?

**Comment by YuriSizov** [comment]
As of commit https://github.com/YuriSizov/boscaceoil-blue/commit/fa405397d52d520dbd1a5a1829f9859069ea80de notes up to 128 units are possible, regardless of the pattern size. I hope this is sufficient for most use cases.

My main concern with allowing arbitrary sizes is that adjusting the note cursor size to large numbers is a painful process. Even at 128 it is going to be frustrating doing it with the mouse. There are also keyboard shortcuts, and I made sure they work by holding them (and added them to the help section, where they were amiss). So it's not going to be totally awful, but still, not that great.

So for that reason I'm going to limit it to 128 for now.

Note that while I've reworked exporters to account for that, it may not always be possible to correctly export Bosca Ceoil song to various legacy formats, like XM, where notes cut off at the end of a pattern. But these were always a bit niche and experimental, so I think that's an okay situation for now.
